test_general/api/test_app.py

```diff
@@ -57,4 +59,21 @@ def test_general_api(example_filename):
         ],
     )
     assert response.status_code == 200
+    assert all(x["metadata"]["filename"] == example_filename for i in response.json() for x in i)
+
     assert len(response.json()) > 0
+
+
+@pytest.mark.parametrize(
+    "example_filename",
+    [
+        "fake-xml.xml",
+    ],
+)
+def test_general_api_raises_error(example_filename):
+    client = TestClient(app)
+    test_file = Path("sample-docs") / example_filename
+    with pytest.raises(ValueError):
```

Comment:
Hmm, if you added back:
```
class UnsupportedFileError(Exception):
    def __init__(self, message):
        self.message = message
        super().__init__(self.message)

DEFAULT_MIMETYPES = "application/pdf,application/msword,image/jpeg,image/png,text/markdown," \
                    "text/x-markdown,application/epub,application/epub+zip,text/html," \
                    "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet," \
                    "application/vnd.ms-excel,application/vnd.openxmlformats-officedocument." \
                    "presentationml.presentation," \
                    "application/vnd.ms-powerpoint,application/xml"


if not os.environ.get("UNSTRUCTURED_ALLOWED_MIMETYPES", None):
    os.environ["UNSTRUCTURED_ALLOWED_MIMETYPES"] =  DEFAULT_MIMETYPES
    
```
from earlier in this branch
to the pipeline-notebook and  rerun`make generate-api`
do we not get the ValueError?

(my prior comment had been about removing the block
```
    if file_content_type and file_content_type not in os.environ.get("UNSTRUCTURED_ALLOWED_MIMETYPES"):
        raise UnsupportedFileError(message=f"{file_content_type} not currently supported")
```
only inside of `pipeline_api` but keeping the rest)

Reply:
ohhh I see, I'll try that.